packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -501,10 +499,11 @@ export class StripeHandler {
         priceId,
         paymentMethodId,
         promotionCode: promotionCodeFromRequest,
-        idempotencyKey,
         metricsContext,
       } = request.payload as Record<string, string>;
 
+      const idempotencyKey = generateIdempotencyKey(uid, priceId);
```

Comment:
This doesn't need to include the payment method id?

Reply:
Ah I forgot to mention this in our previous discussion.

The problem with using `paymentMethodId` is that it'll always be different, since at the moment the front-end always sends a new paymentMethodId for each request.

Thinking a bit more about it, we could use the `paymentMethod.card.fingerprint`, since that should only be different when the card changes.

Let me make the change quick.